tests/RegExLib.FunctionalTests/Api/Expressions/ApiExpressionsControllerList.cs

```diff
@@ -0,0 +1,34 @@
+﻿using RegExLib.Core.Entities;
+using RegExLib.Web;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using RegExLib.Web.Seeds;
+using Xunit;
+
+namespace RegExLib.FunctionalTests.Api.expressions
+{
+  public class ApiExpressionsControllerList : IClassFixture<CustomWebApplicationFactory<Startup>>
+  {
+    private readonly HttpClient _client;
+
+    public ApiExpressionsControllerList(CustomWebApplicationFactory<Startup> factory)
+    {
+      _client = factory.CreateClient();
+    }
+
+    [Fact]
+    public async Task ReturnsPageNumberOneExpressions()
+    {
+      var response = await _client.GetAsync("/api/expressions?page=0");
+      response.EnsureSuccessStatusCode();
+      var stringResponse = await response.Content.ReadAsStringAsync();
+      var result = JsonConvert.DeserializeObject<IEnumerable<Expression>>(stringResponse).ToList();
+
+      Assert.Single(result);
+      Assert.Contains(result, i => i.Title == ExpressionsSeed.Expression1.Title);
+    }
+  }
+}
```

Comment:
You might also consider testing these scenarios:

1. given _totalCount_ > _pageSize_ when _page_ = 0 then should return _pageSize_ elements
2. given totalCount > pageSize when _page_ = 1 then should return results for page 1
3. other boundary conditions
  a. _totalCount_ = 0
  a. _page_ < 0
  b. last page
  c. _page_ > # pages

Reply:
I already change seed class.
Do we need to add more than 25 records in expression seed to make these tests?

> 
> 
> You might also consider testing these scenarios:
> 
>     1. given _totalCount_ > _pageSize_ when _page_ = 0 then should return _pageSize_ elements
> 
>     2. given totalCount > pageSize when _page_ = 1 then should return results for page 1
> 
>     3. other boundary conditions
>        a. _totalCount_ = 0
>        a. _page_ < 0
>        b. last page
>        c. _page_ > # pages

Reply:
I suppose there's a couple options:

1. If page size were made configurable then I think testing the degenerate case where pageSize = 1 would probably be sufficient.

2. If it stays fixed at 25, you would indeed need at least 26. For this option, I might consider generating the seed objects using something like `Enumerable.Range(1, perPage + 1).Select(index => new { FullName = $"FullName{index}"})`